tensorflow/python/training/slot_creator.py

```diff
@@ -42,12 +42,13 @@
 from tensorflow.python.framework import ops
 from tensorflow.python.ops import array_ops
 from tensorflow.python.ops import variables
+from tensorflow.python.ops import variable_scope
 
 
 def _create_slot_var(primary, val, scope):
   """Helper function for creating a slot variable."""
 
-  slot = variables.Variable(val, name=scope, trainable=False)
+  slot = variable_scope.get_variable(scope, initializer=val, trainable=False)
```

Comment:
And I review this code again. If we think `RNN/GRUCell/Gates/Linear/Matrix_N/Adam:0 (150, 200)` is suitable for understanding, we can remove this [change](https://github.com/Syndrome777/tensorflow/blob/06b751fd7c5dbf5ae359156c6bbaf4c78881d72e/tensorflow/python/training/optimizer.py#L592) which add slot_name as suffix (`Adam/m` and `Adam/v`) for better understanding mentioned above.